src/sbvr-api/translations.ts

```diff
@@ -0,0 +1,218 @@
+): void => {
+	for (const [key, relationship] of Object.entries(
+		relationships as RelationshipInternalNode,
+	)) {
+		if (key === '$') {
+			return;
+		}
+
+		let mapping = (relationship as RelationshipLeafNode).$;
+		if (mapping != null && mapping.length === 2) {
+			if (!key.includes('$')) {
+				mapping = _.cloneDeep(mapping);
+				mapping[1]![0] = `${mapping[1]![0]}$${alias}`;
+				(relationships as RelationshipInternalNode)[`${key}$${alias}`] = {
+					$: mapping,
+				};
+				delete (relationships as RelationshipInternalNode)[key];
+			}
+		}
+		namespaceRelationships(relationship, alias);
+	}
+};
+
+export const translateAbstractSqlModel = (
+	fromAbstractSqlModel: AbstractSqlModel,
+	toAbstractSqlModel: AbstractSqlModel,
+	fromVersion: string,
+	toVersion: string,
+	definitions: Dictionary<
+		| (Definition & { $toResource?: string })
+		| Dictionary<string | AliasValidNodeType>
+	> = {},
+): Dictionary<string> => {
+	const isDefinition = (d: (typeof definitions)[string]): d is Definition =>
+		'abstractSql' in d;
+
+	const resourceRenames: Dictionary<string> = {};
+
+	fromAbstractSqlModel.rules = toAbstractSqlModel.rules;
+
+	const fromKeys = Object.keys(fromAbstractSqlModel.tables);
+	const nonexistentTables = _.difference(Object.keys(definitions), fromKeys);
+	if (nonexistentTables.length > 0) {
+		throw new Error(
+			`Tried to define non-existent resources: '${nonexistentTables.join(
+				', ',
+			)}'`,
+		);
+	}
+	for (const [synonym, canonicalForm] of Object.entries(
+		toAbstractSqlModel.synonyms,
+	)) {
+		// Don't double alias
+		if (synonym.includes('$')) {
+			fromAbstractSqlModel.synonyms[synonym] = canonicalForm;
+		} else {
+			fromAbstractSqlModel.synonyms[
+				`${synonym}$${toVersion}`
+			] = `${canonicalForm}$${toVersion}`;
+		}
+	}
+	const relationships = _.cloneDeep(toAbstractSqlModel.relationships);
+	namespaceRelationships(relationships, toVersion);
+	for (let [key, relationship] of Object.entries(relationships)) {
+		// Don't double alias
+		if (!key.includes('$')) {
+			key = `${key}$${toVersion}`;
+		}
+		fromAbstractSqlModel.relationships[key] = relationship;
+	}
+
+	// TODO: We also need to keep the original relationship refs to non $version resources
+	// Also alias for ourselves to allow explicit referencing
+	const aliasedFromRelationships = _.cloneDeep(
+		fromAbstractSqlModel.relationships,
+	);
+	namespaceRelationships(aliasedFromRelationships, fromVersion);
+	for (let [key, relationship] of Object.entries(aliasedFromRelationships)) {
+		// Don't double alias
+		if (!key.includes('$')) {
+			key = `${key}$${fromVersion}`;
+			fromAbstractSqlModel.relationships[key] = relationship;
+		}
+	}
+
+	for (let [key, table] of Object.entries(toAbstractSqlModel.tables)) {
+		// Don't double alias
+		if (!key.includes('$')) {
+			key = `${key}$${toVersion}`;
+		}
+		fromAbstractSqlModel.tables[key] = _.cloneDeep(table);
+	}
+
+	for (const key of fromKeys) {
+		const definition = definitions[key];
```

Comment:
```suggestion		
        const translation = definitions[key];
```
And following, makes it more readiable, at least for me :)